storage/innobase/srv/srv0srv.cc

```diff
@@ -846,13 +846,8 @@ srv_printf_innodb_monitor(
 		static_cast<double>(btr_cur_n_non_sea - btr_cur_n_non_sea_old)
 		/ time_elapsed);
 	btr_cur_n_sea_old = btr_cur_n_sea;
-#else /* BTR_CUR_HASH_ADAPT */
-	fprintf(file,
-		"%.2f non-hash searches/s\n",
-		static_cast<double>(btr_cur_n_non_sea - btr_cur_n_non_sea_old)
-		/ time_elapsed);
-#endif /* BTR_CUR_HASH_ADAPT */
 	btr_cur_n_non_sea_old = btr_cur_n_non_sea;
+#endif /* BTR_CUR_HASH_ADAPT */
```

Comment:
Now that you turned `btr_cur_n_sea` and `btr_cur_n_non_sea` into sharded counters, we’d better read the values only once, because I do not think that the compiler is allowed to optimize away the second read by caching the first one’s result. Something like this:
```c++
const ulint with_ahi = btr_cur_n_sea, without_ahi = btr_cur_n_non_sea;
fprintf(file, … with_ahi … without_ahi);
btr_cur_n_sea_old = with_ahi;
btr_cur_n_non_sea_old = without_ahi;
```
You might also add a comment that the `_old` variables are protected by `srv_innodb_monitor_mutex`.